dss/operations/iam.py

```diff
@@ -0,0 +1,228 @@
+"""
+Obtain information about IAM (policies and permissions) from cloud providers.
+"""
+import os
+import sys
+import select
+import typing
+import argparse
+import json
+import requests
+import logging
+from functools import lru_cache
+
+from botocore.exceptions import ClientError
+
+from dcplib.security import DCPServiceAccountManager
+from dss.operations import dispatch
+from dss.util.aws.clients import iam as iam_client  # type: ignore
+
+"""
+This module is quite long, so here is a table of contents and summary.
+
+Table of Contents:
+- fusillade utility functions
+- functions to dump/list all policies
+    - cloud providers have separate, parallel sets of methods
+    - all cloud providers have an extract method to extract policy information
+    - all cloud providers have wrapper methods to list/dump the information
+- functions to dump/list policies grouped by asset type
+    - an asset is a user, a group, or a role
+    - these functions have a different logical structure from the above
+    - aws has a single function to list policies grouped by asset
+    - fusillade has separate functions for each type of asset: users, groups, and roles
+- dss-ops command line utility functionality
+    - list action
+"""
```

Comment:
This sort of documentation needs to be maintained along with code updates. It's like an implicit dependency.

Consider breaking up this file into smaller, self explanatory modules. For instance, `dss/operations/aws_iam_interface.py`. Such a module does not necessarily need to implement CLI endpoints (e.g. `dss/operations/util.py`). This can be future work.